common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -316,6 +317,7 @@ class @Problem
       switch response.success
         when 'incorrect', 'correct'
           window.SR.readElts($(response.contents).find('.status'))
+          @el.trigger('contentChanged', [@id, response.contents])
```

Comment:
That's little tricky, the whole problem module initializes again on sequence navigation. On every sequence switch, `div #seq_content`(that's just a container) renders selected sequence's content (i.e. this contains problem in descendants)  so your mentioned `el` is going to be selected from this content container. Now, this content is going to be replaced with some other problem on switching to next sequence and `el` will be selected again. 

So, If we update the data attr of current `el`, change will be lost on switching the sequence as whole `el` is going to be newly selected out of another problem.

That's why, I am triggering the event from problem module and updating the data attr in sequence module – as this is where the updated content is going to be available for the problem in first place!